types/gravatar/index.d.ts

```diff
@@ -1,23 +1,35 @@
 // Type definitions for gravatar v1.8.0
 // Project: https://github.com/emerleite/node-gravatar
 // Definitions by: Denis Sokolov <https://github.com/denis-sokolov>
+//                 April Arcus <https://github.com/aprilarcus>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-declare namespace GravatarModule {
-  type Options = {
-    d?: string
-    default?: string
-    f?: string
-    forcedefault?: string
-    format?: string
-    protocol?: string
-    r?: string
-    rating?: string
-    s?: string
-    size?: string
+declare module 'gravatar' {
+
+  type URL = string
+
+  type Default = URL | '404' | 'mp' | 'identicon' | 'monsterid' | 'wavatar' | 'retro' | 'robohash' | 'blank'
+
+  type Rating = 'g' | 'pg' | 'r' | 'x'
+
+  interface Options {
+    size?: number,
+    s?: number,
+    default?: Default,
+    d?: Default,
+    rating?: Rating,
+    r?: Rating,
+    forcedefault?: 'y',
+    f?: 'y',
+    protocol?: boolean | 'http' | 'https',
+    cdn?: string
   }
 
-  function url(email: string, options?: Options, forceProtocol?: boolean): string;
-}
+  interface ProfileOptions extends Options {
```

Comment:
You don't need to change type to an interface to provide extend functionality. With type arithmetic everything can be expressed, too. I don't care much about type vs interface, but I got surprised as to why the type was switched to an interface, seemed unrelated to the PR.

```suggestion
  type ProfileOptions = Options & {
```

Reply:
Sure, we could do that, but I think `interface ... extends` reads more clearly, and the semantics are identical for practical purposes.

Reply:
Considering how they are identical except for a personal matter of preference, wouldn't it make sense to keep whatever convention is already used in the code? :-)

To be clear, I could not care less about syntax conventions in this file, so the debate is purely theoretical for me.

Reply:
If you think `interface ... extends` is aesthetically offensive, I'll change it, but I repeat that I find it more readable. Interfaces are good descriptors for option bags; intersection types are more powerful and in this case, unnecessarily so. Consider the [law of least power](https://en.wikipedia.org/wiki/Rule_of_least_power).

Reply:
I don’t think the `interface` keyword is aesthetically offensive. I do believe that type arithmetic reads more clearly.

But as I’ve said before, I don’t care if you rewrite the entire file with an ASCII graphic of a dinosaur. :-)